src/Tribe/Events/Attendee_List_Display.php

```diff
@@ -0,0 +1,223 @@
+ *
+ * @since   4.12.0
+ *
+ * @package Tribe\Tickets\Events
+ */
+class Attendee_List_Display {
+
+	/**
+	 * @var string Meta name to control whether the Attendee List
+	 *             meta was changed by a shortcode in the content.
+	 */
+	protected static $attendee_list_by_shortcode = 'tribe_tickets_attendee_list_triggered_by_shortcode';
+
+	/**
+	 * Check if given event is hiding the attendees list.
+	 *
+	 * @param int|WP_Post $post The Post being checked.
+	 * @throws InvalidArgumentException Could not determine if given event is hiding or showing the Attendee List.
+	 *
+	 * @return bool True if event is hiding the attendees list. False otherwise.
+	 */
+	public function is_event_hiding_attendee_list( $post ) {
+		// Attendees list is a Plus feature, if ET Plus is not present it will always be hidden.
+		if ( ! class_exists( 'Tribe__Tickets_Plus__Main' ) ) {
+			return true;
+		}
+
+		if ( is_numeric( $post ) ) {
+			$post = WP_Post::get_instance( $post );
+		}
+
+		if ( ! $post instanceof WP_Post ) {
+			/*
+			 * There's no way to apply a sensible default in this scenario.
+			 *
+			 * Let's throw an exception so the client calling this function deals
+			 * with it according to the >> caller << context.
+			 */
+			throw new InvalidArgumentException( '$post should be either a post ID or a WP_Post object.' );
+		}
+
+		return (bool) Tribe__Tickets_Plus__Attendees_List::is_hidden_on( $post );
+	}
+
+	/**
+	 * @param bool $should_hide Whether the optout form should be hidden or not.
+	 * @filter tribe_tickets_plus_hide_attendees_list_optout 10 1
+	 * @see    \Tribe\Tickets\Events\Events_Service_Provider::hooks
+	 *
+	 * @return bool
+	 */
+	public function should_hide_optout( $should_hide ) {
+		try {
+			global $post;
+
+			return $this->is_event_hiding_attendee_list( $post );
+		} catch ( Exception $e ) {
+			// Eg: global $post not a WP_Post object.
+			return $should_hide;
+		}
+	}
+
+	/**
+	 * Determines whether this post is displaying the Attendees List.
+	 *
+	 * @param int|WP_Post $post The Post being checked.
+	 * @action save_post_tribe_events 10 1
+	 * @see    \Tribe\Tickets\Events\Events_Service_Provider::hooks
+	 *
+	 * @retyrn void
+	 */
+	public function maybe_update_attendee_list_hide_meta( $post ) {
+		// Attendees list is a Plus feature, if ET Plus is not present we don't have to update the meta.
+		if ( ! class_exists( 'Tribe__Tickets_Plus__Main' ) ) {
+			return;
+		}
+
+		// Early bail: is an autosave or auto-draft.
+		if ( wp_is_post_autosave( $post ) || wp_is_post_revision( $post ) ) {
+			return;
+		}
+
+		if ( ! $post instanceof WP_Post ) {
+			$post = get_post( $post );
+		}
+
+		// Early bail: Invalid post.
+		if ( ! $post instanceof WP_Post ) {
+			return;
+		}
+
+		$post_types_allowed_to_have_tickets = (array) tribe_get_option( 'ticket-enabled-post-types', [] );
+
+		// Early bail: This post type can't have tickets
+		if ( ! in_array( $post->post_type, $post_types_allowed_to_have_tickets ) ) {
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 36ab8446e8751a21a2d825b2cd980e90</sub>